dbt/include/athena/macros/materializations/models/table/create_table_as.sql

```diff
@@ -145,16 +145,21 @@
 
 {%- endmacro %}
 
-{% macro safe_create_table_as(temporary, relation, sql) -%}
+{% macro safe_create_table_as(temporary, relation, sql, batch_inserts) -%}
     {%- if temporary -%}
       {%- do run_query(create_table_as(temporary, relation, sql, True)) -%}
       {%- set query_result = relation ~ ' as temporary relation without partitioning created' -%}
     {%- else -%}
-      {%- set query_result = adapter.run_query_with_partitions_limit_catching(create_table_as(temporary, relation, sql)) -%}
-      {%- do log('QUERY RESULT: ' ~ query_result) -%}
-      {%- if query_result == 'TOO_MANY_OPEN_PARTITIONS' -%}
-        {%- do create_table_as_with_partitions(temporary, relation, sql) -%}
-        {%- set query_result = relation ~ ' with many partitions created' -%}
+      {%- if not batch_inserts -%}
+        {%- set query_result = adapter.run_query_with_partitions_limit_catching(
+              create_table_as(temporary, relation, sql)
+            )
+        -%}
+        {%- do log('QUERY RESULT: ' ~ query_result) -%}
```

Comment:
I think previous implementation should be inserted into `else` branch here

Reply:
I believe that it will work also without, but indeed having a if/elif branch might be cleaner to read.

Reply:
Yep, i change that.
Please review.